src/bot/lib/query.js

```diff
@@ -30,18 +30,19 @@ export function findSkillByName(token, search) {
 }
 
 // Find the best match for the given search term, and complain if necessary.
-export function findSkillAndHandleErrors(token, search) {
+export function findItemAndHandleErrors(token, search, queryFunction) {
```

Comment:
I'm pretty sure you could capture the missing error handling logic like so: 

```js
function throwIfMatchErrors(results) {
  const {output, errors} = results;
  if (errors.length) {
    throw abort(...output, ...errors);
  }
  return results;
}
```

...and have the following userland code:

```js
query.categoryByName({token, search})
  .then(parseMatches)
  .then(prepareMatchOutput)
  .then(throwIfMatchErrors)
```

...but it took me like 30 minutes to build up the mental context for this code this morning and I can't dive in again right now.